tawazi/_dag/dag.py

```diff
@@ -287,12 +329,12 @@ def _add_missing_deps(candidate_id: Identifier, xn_ids: Set[Identifier]) -> None
         #  maybe the composed DAG will modify them (e.g. change their tags)
         #  and we don't want to modify the original DAG
         # we avoid adding the inputs because they will be modified just afterwards
-        xn_dict = {
-            in_id: deepcopy(self.exec_nodes[in_id]) for in_id in set_xn_ids if in_id not in in_ids
-        }
+        xn_dict = StrictDict(
+            (in_id, deepcopy(self.exec_nodes[in_id])) for in_id in set_xn_ids if in_id not in in_ids
+        )
 
         # change input ExecNodes to ArgExecNodes
-        new_in_ids = [make_axn_id(old_id, id_="composed") for old_id in in_ids]
+        new_in_ids = [make_axn_id("composed", old_id) for old_id in in_ids]
```

Comment:
maybe make this string literal a twz constant ?

Reply:
this reminded of using qualname instead of composed which is more adapted to this case